src/App.tsx

```diff
@@ -1,26 +1,506 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  deleteTodo,
+  getTodos,
+  addTodo,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Error } from './components/ErrorMessage/ErrorMessage';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [query, setQuery] = useState<string>('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [itemsLeft, setItemsLeft] = useState<number>(0);
```

Comment:
General rule - do not create state if you can compute it based on other state or props values